### PR TITLE
Add migration/codemod to replace motion custom properties

### PR DESCRIPTION
**Pull request by lgriffee**
### WHY are these changes introduced?

Part of https://github.com/Shopify/polaris/issues/8405

### WHAT is this pull request doing?

This PR creates a migration and codemod for deprecated custom properties in v11 within the motion token group using the generic codemod [`styles-replace-custom-property`](https://github.com/Shopify/polaris/pull/8265). It also adds the color migration to the new codemod folder. 

#### Motion

| Deprecated Token             | Replacement Value                   |
| ---------------------------- | ----------------------------------- |
| `--p-linear`                 | `--p-motion-linear`                 |
| `--p-ease-in-out`            | `--p-motion-ease-in-out`            |
| `--p-ease-out`               | `--p-motion-ease-out`               |
| `--p-ease-in`                | `--p-motion-ease-in`                |
| `--p-ease`                   | `--p-motion-ease`                   |
| `--p-duration-0`             | `--p-motion-duration-0`             |
| `--p-duration-50`            | `--p-motion-duration-50`            |
| `--p-duration-100`           | `--p-motion-duration-100`           |
| `--p-duration-150`           | `--p-motion-duration-150`           |
| `--p-duration-200`           | `--p-motion-duration-200`           |
| `--p-duration-250`           | `--p-motion-duration-250`           |
| `--p-duration-300`           | `--p-motion-duration-300`           |
| `--p-duration-350`           | `--p-motion-duration-350`           |
| `--p-duration-400`           | `--p-motion-duration-400`           |
| `--p-duration-450`           | `--p-motion-duration-450`           |
| `--p-duration-500`           | `--p-motion-duration-500`           |
| `--p-duration-5000`          | `--p-motion-duration-5000`          |
| `--p-keyframes-bounce`       | `--p-motion-keyframes-bounce`       |
| `--p-keyframes-fade-in`      | `--p-motion-keyframes-fade-in`      |
| `--p-keyframes-pulse`        | `--p-motion-keyframes-pulse`        |
| `--p-keyframes-spin`         | `--p-motion-keyframes-spin`         |
| `--p-keyframes-appear-above` | `--p-motion-keyframes-appear-above` |
| `--p-keyframes-appear-below` | `--p-motion-keyframes-appear-below` |

### CODE REVIEW

File: .changeset/stupid-meals-hear.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris-codemods': minor
+---
+
+Created a migration to replace deprecated `motion` custom properties in polaris-react v11.0.0
```
Comment (aaronccasanova) [inline .changeset/stupid-meals-hear.md]:
```suggestion
Created a migration to replace deprecated `motion` custom properties in `@shopify/polaris` v11.0.0
```

File: .changeset/stupid-meals-hear.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris-codemods': minor
```
Comment (lgriffee) [inline .changeset/stupid-meals-hear.md:2]:
```suggestion
'@shopify/polaris-codemods': minor
'@shopify/polaris-migrator': minor
```

**Review by aaronccasanova — APPROVED** [review]

**Review by alex-page — APPROVED** [review]

**Review by sam-b-rose — APPROVED** [review]
This looks excellent–nice work contributing this to the new `polaris-codemods` package!

We will need to copy this over to the `polaris-migrator` package as well since `polaris-codemods` package can't be run as a standalone CLI. The runner logic has been moved to the Shopify CLI. We are very close to having the feature shipped in the CLI, but until then we will still need to use the `polaris-migrator` package.

Copying this migration over to that package shouldn't require many adjustments, but let me know if you need help 👍 if this migration isn't urgent, we could wait til the Shopify CLI transform runner is ready

**Comment by lgriffee** [comment]
> This looks excellent–nice work contributing this to the new polaris-codemods package!
> 
> We will need to copy this over to the polaris-migrator package as well since polaris-codemods package can't be run as a standalone CLI. The runner logic has been moved to the Shopify CLI. We are very close to having the feature shipped in the CLI, but until then we will still need to use the polaris-migrator package.
> 
> Copying this migration over to that package shouldn't require many adjustments, but let me know if you need help 👍 if this migration isn't urgent, we could wait til the Shopify CLI transform runner is ready

Awesome, thanks Sam! There should be a copy of both motion and color in both folders after a92b9d3!

**Pull request merged**